src/js/components/Marathon.jsx

```diff
@@ -44,21 +46,24 @@ var Marathon = React.createClass({
   },
 
   componentDidMount: function () {
-    this.onRouteChange();
+    PluginActions.requestPlugins();
 
+    this.onRouteChange();
     this.bindKeyboardShortcuts();
 
-    PluginStore.once(PluginEvents.BOOTSTRAP_COMPLETE, () => {
-      this.startPolling();
-    });
-
-    PluginStore.bootstrap();
+    PluginStore.on(PluginEvents.CHANGE, this.onPluginStoreChange);
   },
 
   componentWillReceiveProps: function () {
     this.onRouteChange();
   },
 
+  onPluginStoreChange: function () {
+    if (PluginStore.getPluginLoadingState() > States.STATE_LOADING) {
```

Comment:
OK, cool. 
Only one question. How could one plugin determine that another plugin is loaded?